core/src/main/java/io/grpc/internal/ServerImplBuilder.java

```diff
@@ -44,62 +105,234 @@
+            censusStatsAccessor.getDeclaredMethod(
+                "getServerStreamTracerFactory",
+                boolean.class,
+                boolean.class,
+                boolean.class);
+        censusStatsTracerFactory =
+            (ServerStreamTracer.Factory) getServerStreamTracerFactoryMethod
+                .invoke(
+                    null,
+                    recordStartedRpcs,
+                    recordFinishedRpcs,
+                    recordRealTimeMetrics);
+      } catch (ClassNotFoundException e) {
+        // Replace these separate catch statements with multicatch when Android min-API >= 19
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (NoSuchMethodException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (IllegalAccessException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (InvocationTargetException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      }
+      if (censusStatsTracerFactory != null) {
+        tracerFactories.add(censusStatsTracerFactory);
+      }
+    }
+    if (tracingEnabled) {
+      ServerStreamTracer.Factory tracingStreamTracerFactory = null;
+      try {
+        Class<?> censusTracingAccessor =
+            Class.forName("io.grpc.census.InternalCensusTracingAccessor");
+        Method getServerStreamTracerFactoryMethod =
+            censusTracingAccessor.getDeclaredMethod("getServerStreamTracerFactory");
+        tracingStreamTracerFactory =
+            (ServerStreamTracer.Factory) getServerStreamTracerFactoryMethod.invoke(null);
+      } catch (ClassNotFoundException e) {
+        // Replace these separate catch statements with multicatch when Android min-API >= 19
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (NoSuchMethodException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (IllegalAccessException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      } catch (InvocationTargetException e) {
+        log.log(Level.FINE, "Unable to apply census stats", e);
+      }
+      if (tracingStreamTracerFactory != null) {
+        tracerFactories.add(tracingStreamTracerFactory);
+      }
+    }
+    tracerFactories.addAll(streamTracerFactories);
+    tracerFactories.trimToSize();
+    return Collections.unmodifiableList(tracerFactories);
+  }
+
   public InternalChannelz getChannelz() {
-    return super.getChannelz();
+    return channelz;
   }
 
-  @Override
+  /**
+   * Transport implementors must implement {@link ClientTransportServersBuilder} to transport
+   * specific information for the server. This method is mean for Transport implementors and should
+   * not be used by normal users.
+   *
+   * @param streamTracerFactories an immutable list of stream tracer factories
+   */
+  @VisibleForTesting
+  List<? extends InternalServer> buildTransportServers(
```

Comment:
I wouldn't expect any transports to need this method and I don't see any references. It seems it could be made private or inlined into build(). If you want to keep it for ServerImplBuilderTest, then the javadoc comment isn't really appropriate.

Reply:
I don't think that single test valuable anymore, so I went with removing the test, and inlining `buildTransportServers` into `build`.